ITK guidelines

rdebroiz opened the pull request:
This should fix #4, #3 and #2

Commit:
Update TBBImageToImageFilter.remote.cmake

Commit:
STYLE: Add ITK copyright notice at the begining of C++ files.

Commit:
DOC: Add missing DOCUMENTATION variable to CMake.

Commit:
STYLE: Add helper class inside TBBImageToImageFilterTest and prune useless files.

Commit:
STYLE: Follow ITK guidelines.

Commit:
STYLE: add ingroup markup to documentation.

Commit:
STYLE: fix statement within function indentation.

Commit:
STYLE: fix statement within function indentation (again).

rdebroiz commented:
Note: in order to have the Doxygen test to succeed, I had to add a `\ingroup` markup to the documentation, I set it to `ITKTBBImageToImageFilter` for now. I don't now what would be correct.

Commit:
Add PrintSelf function to TBBImageToImageFilter.

jhlegarreta commented:
Since the module dwells and would be downloaded to the Remote folder/group, the right one would be `Remote`.

Commit:
STYLE: more accordance to itk style guidlines.

Move Some Doxygen strings to .h
move TBBFunctor declaration to .h
Replace C-Cast by static_cast

Commit:
DOC: change group to Remote for doxygen.

Commit:
STYLE: reformat doxygen strings.

Commit:
DOC: Set back \ingroup to TBBImageToImageFilter.

Commit:
DOC: Set back \ingroup to TBBImageToImageFilter.

rdebroiz commented:
I had to put back `ITKTBBImageToImageFilter` as `\ingroup` or i had the following ctest error:
```
error: "\ingroup TBBImageToImageFilter" not set in class TBBImageToImageFilter.
error: "\ingroup TBBImageToImageFilter" not set in class TBBFunctor.
```

Commit:
STYLE: Remove useless header: test/itkTBBTestFilter.h

Commit:
STYLE: Use ITK macro, and renaming.
NbJobs -> NumberOfJobs, use itkGetConstMacro and itkSetMacro
NbReduceDimensions -> NuumberOfDimensionToreduce use itkGetConstMacro
Add typedf JobIdType and DimensionReductionType
move ITK_DISALLOW_COPY_AND_ASSIGN at the top of the private section
correct typo: artical -> article

Commit:
Fix issue #5 'Switching ITK_USE_TBB from ON to OFF'

Explicitly unset some CMake vairiable if ITK_USE_TBB is not defined.
Remove -DITK_USE_TBB flag if ITK_USE_TBB is not defined.

Commit:
Merge pull request #6 from rdebroiz/cmake-itk_not_use_tbb

Fix issue #5 'Switching ITK_USE_TBB from ON to OFF'

Commit:
Update CMakeLists.txt

Add a TODO

Commit:
Update CMakeLists.txt

Add a coment

Commit:
Add missing code for ResetJobQueue.

Commit:
ENH: Adequate module tests to ITK test driver structure.

Use the itkTestDriver to set up the required environment for the test
suite to be run under the ITK testing framework.

Make the necessary modifications to the CMakeLists.txt file.

Rename the test file main method name to adapt to these changes.

Commit:
ENH: Improve PrintSelf method implementation.

Print all ivars.

Use the NumericTraits::PrintSelf method to appropriately print
typedef'ed ivars.

Call the Superclass PrintSelf.

Commit:
STYLE: A few style cleanups.

Make the itk namespace brace dwell on a line of its own.

Remove unnecessary method doc from the implementation file.

Indent appropriately method documentation.

Add white spaces between the assignment operator and the assigned value.

Commit:
Fix itkTBBImageToImageFilterTest parameters.

Commit:
Merge pull request #8 from jhlegarreta/StyleCleanUps

STYLE: A few style cleanups.

Commit:
Merge pull request #7 from jhlegarreta/ImprovePrintSelfImplementation

ENH: Improve PrintSelf method implementation.

Commit:
FIX: Add missing #endif

Commit:
Fix: name changes.

Commit:
Merge pull request #9 from jhlegarreta/AdequateTestToITKTestDriverStructure

ENH: Adequate module tests to ITK test driver structure.

Commit:
Replace JOB_PER_THREAD_RATIO by a static member.

Commit:
Merge pull request #11 from rdebroiz/JobPerThreadRatio_StaticMember

Replace JOB_PER_THREAD_RATIO by a static member.
